react/features/analytics/middleware.js

```diff
@@ -12,25 +78,81 @@ import { initAnalytics, resetAnalytics } from './functions';
  * @returns {Function}
  */
 MiddlewareRegistry.register(store => next => action => {
-    switch (action.type) {
-    case SET_CONFIG: {
+    if (action.type === SET_CONFIG) {
         if (navigator.product === 'ReactNative') {
             // Reseting the analytics is currently not needed for web because
             // the user will be redirected to another page and new instance of
             // Analytics will be created and initialized.
             resetAnalytics();
         }
+    }
 
+    const result = next(action);
+
+    switch (action.type) {
+    case CONFERENCE_JOINED: {
+        const { dispatch, getState } = store;
+        const state = getState();
+
+        dispatch({
+            type: UPDATE_LOCAL_TRACKS_DURATION,
```

Comment:
Don't we usually abstract the action type in an action creator (actions.js)?

Reply:
Most of the times, but we also have a lot of actions that don't have action creator. In this case I didn't use an action creator because it is relatively simple one and it is used only in the same feature.

If you insist on creating one or if we have such a rule I can add one. Do you want me to add?

Reply:
`we also have a lot of actions that don't have action creator.` Example?

Reply:
when I grep jitsi-meet for `dispatch({` I see 39 results. All those don't have action creator.
Although most of them are used in actions.js files only....

Do you want me to create an action creator?